test/Elastic.Apm.Tests/MetricsTests.cs

```diff
@@ -36,6 +36,7 @@ public void SystemCpu()
 			if (!RuntimeInformation.IsOSPlatform(OSPlatform.Linux) && !RuntimeInformation.IsOSPlatform(OSPlatform.Windows)) return;
 
 			var systemTotalCpuProvider = new SystemTotalCpuProvider();
+			systemTotalCpuProvider.GetSamples(); //on Windows the perf. counters sometimes return 0 the first time -> we call it 2 times
```

Comment:
According to https://stackoverflow.com/questions/2181828/why-the-cpu-performance-counter-kept-reporting-0-cpu-usage CPU % counter always returns 0 on the first read. You can do the first read in constructor the same way it’s done for Linux.

Reply:
Can we guess why this happens? 😆

Reply:
👍  moved the 1. call to `.ctor`.

Reply:
To me it just proves that it's not the best design - even those of us that wanted to do the same in our code did not expect other libraries to do it :)